### ISSUE TITLE
failed to configure msgpack with cmake

### ISSUE THREAD

**Issue by hg-zt** [microsoft/vcpkg#7205]
<details>
  <summary>After successful installation of msgpack:</summary>

```
$ ./vcpkg install msgpack

The following packages will be built and installed:
    msgpack[core]:x64-linux
Starting package 1/1: msgpack:x64-linux
Building package msgpack[core]:x64-linux...
-- Using cached /home/hg/b/vcpkg/downloads/msgpack-msgpack-c-cpp-3.1.1.tar.gz
-- Using source at /home/hg/b/vcpkg/buildtrees/msgpack/src/cpp-3.1.1-eab3c56b41
-- Applying patch /home/hg/b/vcpkg/ports/msgpack/add-static-lib-option.patch
-- Applying patch failed. This is expected if this patch was previously applied.
-- Configuring x64-linux-dbg
-- Configuring x64-linux-rel
-- Building x64-linux-dbg
-- Building x64-linux-rel
-- Performing post-build validation
-- Performing post-build validation done
Building package msgpack[core]:x64-linux... done
Installing package msgpack[core]:x64-linux...
Installing package msgpack[core]:x64-linux... done
Elapsed time for package msgpack:x64-linux: 2.423 s

Total elapsed time: 2.423 s

The package msgpack:x64-linux provides CMake targets:

    find_package(msgpack CONFIG REQUIRED)
    target_link_libraries(main PRIVATE msgpackc msgpackc-cxx msgpackc-static)


```

</details>

<details>
  <summary>Configure the project with a minimal `CMakeLists.txt`:</summary>

```cmake
cmake_minimum_required(VERSION 3.10)

project(test-mp)

find_package(msgpack CONFIG REQUIRED)

add_executable(test-mp main.cpp)

target_link_libraries(test-mp PRIVATE msgpackc msgpackc-cxx msgpackc-static)
```

</details>

using the toolchain
```
$ cmake -DCMAKE_TOOLCHAIN_FILE=~/b/vcpkg/scripts/buildsystems/vcpkg.cmake ..

CMake Error at /home/hg/b/vcpkg/scripts/buildsystems/vcpkg.cmake:198 (_add_library):
  _add_library cannot create ALIAS target "msgpackc" because target
  "msgpackc-static" is imported but not globally visible.
Call Stack (most recent call first):
  /home/hg/b/vcpkg/installed/x64-linux/share/msgpack/msgpack-config.cmake:45 (add_library)
  /home/hg/b/vcpkg/scripts/buildsystems/vcpkg.cmake:266 (_find_package)
  CMakeLists.txt:5 (find_package)


-- Configuring incomplete, errors occurred!
See also "/tmp/test-mp/build/CMakeFiles/CMakeOutput.log".
```

I could use all other required packages from `vcpkg` but msgpack. 

CMake Version: 3.13.2

**Comment by Voskrese** [comment]
> `$ cmake -DCMAKE_TOOLCHAIN_FILE=~/b/vcpkg/scripts/buildsystems/vcpkg.cmake ..`

triplet?

**Comment by pslacerda** [comment]
It also happened to me.

I tried to update to tag `cpp-3.2.0` and skip the patches but the error was the same.

**Comment by ChisholmKyle** [comment]
I'm also having this problem with a fresh checkout of tags/cpp-3.2.0

**Comment by giellamoswhard** [comment]
Same Here

**Comment by giellamoswhard** [comment]
I've temporarly got around this by adding GLOBAL to the the line
`
add_library(msgpackc-static STATIC IMPORTED GLOBAL)`

in the file vcpkg/installed/x64-linux/share/msgpack/msgpack-targets.cmake

**Comment by usagi** [comment]
Mark to the related issue on `msgpack-c`: https://github.com/msgpack/msgpack-c/issues/801

**Comment by raffber** [comment]
I think this can be closed now. This was fixed with #13441.

**Comment by PhoebeHui** [comment]
@raffber, thanks for your comments!
